slides/src/main/asciidoc/lecture.adoc

```diff
@@ -386,19 +484,44 @@ digraph G {
 
 graph [ dpi = 180 ];
 rankdir="LR";
-A1[label="<STOP>"];
-B1[label="<STOP>"];
 
 Kafka;
 
-Kafka[shape=box,label="reduce \n (a, b) → <STOP>"]
+Kafka[shape=box,label="reduce \n (l, r) → <STOP>"]
 
 Kafka -> Cache
 
 Cache[shape=box,label=<<table cellspacing="5" cellborder="0"  border="0">
         <tr><td colspan="2">Cache</td></tr>
         <tr><td>&lt;STOP&gt;</td>
         <vr/><td>&lt;STOP&gt;</td></tr></table>>]
+
+
+}
+----
+
+== Why it's not working
+
+
+[graphviz,"cache3b.png"]
```

Comment:
For these last two slides, I added a separate step for consuming the last input into the cache (`cache3.png`) and then flushing the cache, emitting the stop,stop results (`cache3b.png`).